### PR TITLE
Added design for telemetry service sql

### PR DESCRIPTION
**Pull request by BtXin**
**What this PR does / why we need it**:
Design for telemetry service sql server
**Will this PR make the community happier**? 
Yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [ ] unit test
- [ ] e2e test
- [x] other (please specify)
Doc

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Review by leepala — APPROVED** [review]

File: docs/design/shifu-framework/telemetry-service-SQLServer_MySQL.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+## Design Non-Goal
+- Let telemetry service support any random endpoints
+- Let telemetry service serve as SQL service.
+
+## Design Details
+We will predefine the schema for the database, since the uncertainty of schema and data format will make the the feature hard to use and debug.
+
+The predefined schema looks like:
+```SQL
+CREATE TABLE DeviceMeasurements
```
Comment (tomqin93) [inline docs/design/shifu-framework/telemetry-service-SQLServer_MySQL.md:18]: use a more generic name? i.e. Shifu Telemetry or let user configure the table name

File: docs/design/shifu-framework/telemetry-service-SQLServer_MySQL.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+(
+    TelemetryID INT IDENTITY(1,1) PRIMARY KEY,
+    DeviceID INT,
+    TelemetryData TEXT,
+    TelemetryTimeStamp DATETIME
```
Comment (tomqin93) [inline docs/design/shifu-framework/telemetry-service-SQLServer_MySQL.md:23]: should we just omit the Telemetry in `TelemetryData` and `TelemetryTimeStamp`?

**Review by tomqin93 — COMMENTED** [review]
@BtXin added some comment, please take a look

File: docs/design/shifu-framework/telemetry-service-SQLServer_MySQL.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+The predefined schema looks like:
+```SQL
+CREATE TABLE ${DbTableName}
+(
+    TelemetryID INT IDENTITY(1,1) PRIMARY KEY,
+    DeviceName VARCHAR,
+    TelemetryData TEXT,
+    TelemetryTimeStamp DATETIME
+);
```
Comment (Yunxe) [inline docs/design/shifu-framework/telemetry-service-SQLServer_MySQL.md:24]: modified the schema as followed

SQLServer:
```SQL
CREATE TABLE ${DbTableName}
(
    TelemetryID INT IDENTITY(1,1) PRIMARY KEY,
    DeviceName VARCHAR(255),
    TelemetryData TEXT,
    TelemetryTimeStamp DATETIME
);
```
MySQL:
```sql
CREATE TABLE ${DbTableName} (
    TelemetryID INT AUTO_INCREMENT PRIMARY KEY,
    DeviceName VARCHAR(255),
    TelemetryData TEXT,
    TelemetryTimeStamp DATETIME
);
```

File: docs/design/shifu-framework/telemetry-service-SQLServer_MySQL.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    }
+    defer db.Close()
+    ...
+
+}
+
+func insertIntoDB() {
+    ...
+    // DeviceId is retrieved from edgeDeviceSpec, primary id is generated by database
+    result, err := db.DB.Exec(fmt.Sprintf("Insert Into %s Values('%s','%s','%s')", *db.Settings.DBTable, deviceID, rawData, time.Now()))
```
Comment (Yunxe) [inline docs/design/shifu-framework/telemetry-service-SQLServer_MySQL.md:72]: the implemented query
```go
	result, err := db.DB.Exec(fmt.Sprintf("Insert Into %s (DeviceName,TelemetryData,TelemetryTimeStamp) Values('%s','%s','%s')", *db.Settings.DBTable, deviceName, string(rawData), time.Now().Format("2006-01-02 15:04:05")))
```

**Review by tomqin93 — COMMENTED** [review]
@BtXin let's update the design and merge it

**Review by leepala — APPROVED** [review]
LGMT

**Review by kris21he — APPROVED** [review]

**Pull request merged**